sources/packages/backend/apps/queue-consumers/src/services/application/application.service.ts

```diff
@@ -239,4 +248,67 @@ export class ApplicationService {
       })
       .getMany();
   }
+
+  /**
+   * Retrieves second disbursements eligible for a specific notification (likely disbursement
+   * date has passed and the institution has not yet approved the second COE/disbursement).
+   * This method applies several criteria to filter eligible disbursements:
+   * - There's a pending disbursement schedule
+   * - Application status is completed
+   * - No notification of this type has been sent for this assessment
+   * - Disbursement date has passed today
+   * @returns An array of eligible disbursements with relevant details for notification.
+   */
+  async getSecondDisbursementsStillPending(): Promise<
+    SecondDisbursementStillPending[]
+  > {
+    // Sub query to defined if a notification was already sent to the current assessment.
+    const notificationExistsQuery = this.notificationRepo
+      .createQueryBuilder("notification")
+      .select("1")
+      .where("notification.notificationMessage.id = :messageId")
+      .andWhere(
+        "notification.metadata->>'assessmentId' = assessment.id :: text",
+      )
+      .getQuery();
+    // Sub query to retrieve second disbursements as a with disbursement date has
+    // passed today and notification has not been sent for completed applications.
+    const secondDisbursementsQuery = this.disbursementScheduleRepo
+      .createQueryBuilder("disbursement")
+      .select("MAX(disbursement.id)", "id")
+      .innerJoin("disbursement.studentAssessment", "assessment")
+      .innerJoin("assessment.application", "application")
+      .where(`NOT EXISTS (${notificationExistsQuery})`)
+      .andWhere("application.applicationStatus = :applicationStatusCompleted")
+      .andWhere("application.isArchived = false")
+      .andWhere("disbursement.disbursementDate < CURRENT_DATE")
+      .andWhere("disbursement.coeStatus IN (:...coeStatus)")
+      .groupBy("assessment.id")
+      .having("COUNT(assessment.id) = 2")
+      .getQuery();
+    // Return second disbursements still pending.
+    return this.disbursementScheduleRepo
```

Comment:
This query should also check for the COE status as `required` since the disbursement can be pending due to many other possible booking conditions.